hmcts/pre-api
# S28 3223: Custom Health Indicator (Add checks for MK storage account connections)

### JIRA ticket(s)

- S28-3223

> [!IMPORTANT]
> @lucas-phillips28 Before merge need to add `api-MK-SUBSCRIPTION-ID` and `api-MK-CUSTOMER-ID` secrets to key vault in prod + demo

### Change description
- Add custom health indicator which checks storage account connections for MediaKind.
- Add error logs when MediaKind is not connected to storage accounts corresponding to env
- Add details to health endpoint response which shows connection status for each storage account




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Call the /health endpoint and see details under preApi in response object. `{"preApi": { "mediakindConnections": { "preingestsastg": true, "prefinalsastg": true }}, ...}`
